enterprise/api/v1/mixins.py

```diff
@@ -51,22 +51,29 @@ def update_course(self, course, enterprise_customer, enterprise_context):
         Returns:
             (dict): Updated course metadata
         """
+        enterprise_utm_context = {
+            'utm_medium': 'enterprise',
+            'utm_source': enterprise_customer.name
+        }
+
         course['course_runs'] = self.update_course_runs(
             course_runs=course.get('course_runs') or [],
             enterprise_customer=enterprise_customer,
             enterprise_context=enterprise_context,
+            enterprise_utm_context=enterprise_utm_context
         )
 
         # Update marketing urls in course metadata to include enterprise related info (i.e. our global context).
         marketing_url = course.get('marketing_url')
         if marketing_url:
-            course.update({"marketing_url": utils.update_query_parameters(marketing_url, enterprise_context)})
+            query_parameters = dict(enterprise_context, **enterprise_utm_context)
+            course.update({'marketing_url': utils.update_query_parameters(marketing_url, query_parameters)})
 
         # Finally, add context to the course as a whole.
         course.update(enterprise_context)
         return course
 
-    def update_course_runs(self, course_runs, enterprise_customer, enterprise_context):
+    def update_course_runs(self, course_runs, enterprise_customer, enterprise_context, enterprise_utm_context):
```

Comment:
We do not need the new argument, we can add/remove items in the `enterprise_context` dict.

Reply:
see the above comment